src/ProtonPlayer.js

```diff
@@ -84,19 +98,42 @@ export default class ProtonPlayer {
     const [nextTrack] = this.playlist.forward();
     this.player.playNext(nextTrack);
 
-    return this.player.playTrack(this.playlist.current());
+    return this.player.playTrack(this.playlist.current()).then(() => {
+      this.state = PlaybackState.PLAYING;
+      this._fire('state_changed', this.state);
+    });
+  }
+
+  toggle() {
+    debug('ProtonPlayer#toggle');
+
+    console.log(this.state);
+
+    if (this.state === PlaybackState.PLAYING) {
+      return this.pause();
+    } else if (this.state === PlaybackState.PAUSED) {
```

Comment:
No need for `else` after `return`

Reply:
True, but leaving them in place makes the control flow more clear here in my opinion.